File: java/src/main/java/codingdojo/Yatzy.java
package codingdojo;

import java.util.Arrays;

public class Yatzy {

    public static int chance(int... dice) {
        return Arrays.stream(dice).sum();
    }

    public static int yatzy(int... dice) {
        return Arrays.stream(dice).distinct().count() == 1 ? 50 : 0;
    }
}

File: python/yatzy.py
def chance(dice):
    return sum(dice)


def yatzy(dice):
    distinct = set(dice)
    return 50 if len(distinct) == 1 else 0


def dice_frequency(number, dice):
    return len([d for d in dice if d == number])


def all_dice_frequencies(dice):
    return {n: dice_frequency(n, dice)
            for n in (1, 2, 3, 4, 5, 6)}
